### PR TITLE
Add GCSE subject autocomplete

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

This is part of the structured pre-degree qualifications epic and deals with adding GCSE subjects autocomplete.

When a candidate adds an OtherQualification and selects that it is a GCSE, the subject field should use autocomplete on a list of subjects defined in the prototype.

## Changes proposed in this pull request

- Add GCSE subject data to an initializer
- Pass the data into the autocomplete when the OtherQualification is a GCSE
- Add a few lines to the system spec to ensure the correct data is being passed for A levels and GCSEs

image

## Guidance to review

Does this cover everything?

## Link to Trello card

https://trello.com/c/r03kfFrM/2551-dev-structure-pre-degree-qualifications-gcse-subjects

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: config/initializers/qualifications/gcse.rb [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+  'Additional Mathematics',
+  'Additional Science',
+  'Additional Science (Applied)',
+  'Additional Science (General)',
+  'Additional Science (Modular)',
+  'Additional Science A',
+  'Additional Science B',
+  'Agriculture and Land Use',
+  'Ancient History',
+  'Ancient History (Short Course)',
```
Comment (paulrobertlloyd) [inline config/initializers/qualifications/gcse.rb:13]: Can we replace `Short Course` with `Short course` (lowercase ‘c’). This then follows the casing we’re using for ‘Double award’.

**Comment by paulrobertlloyd** [comment]
Separate issue to this, but a GCSE grade isn’t validated… is that to come?

File: config/initializers/qualifications/gcse.rb (hunk 1 above)
Comment (davidgisbey) [reply config/initializers/qualifications/gcse.rb]: Have done

**Comment by davidgisbey** [comment]
@paulrobertlloyd yes that's in the following card https://trello.com/c/jFRrCczq/2552-dev-structured-pre-degree-qualifications-gcse-grades

**Review by paulrobertlloyd — APPROVED** [review]

**Pull request merged**